sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -188,6 +189,15 @@ abstract class BaseSessionStateBuilder(
         customCheckRules
 
     override protected def lookupCatalog(name: String): CatalogPlugin = session.catalog(name)
+
+    override def checkAnalysis(plan: LogicalPlan): Unit = {
+      // We should check it's innerChildren for InsertIntoDataSourceDirCommand
+      val planToCheck = plan match {
+        case e: InsertIntoDataSourceDirCommand => e.query
+        case _ => plan
+      }
+      super.checkAnalysis(planToCheck)
```

Comment:
Can we resolve this issue in the `InsertIntoDataSourceDirCommand` side?

Reply:
@maropu  Yes, I agree that resolving this issue inside `InsertIntoDataSourceDirCommand` is better, but as I have already questioned is the jira: https://issues.apache.org/jira/browse/SPARK-28195, I'm not sure whether there are some special consideration for making the children of `InsertIntoDataSourceDirCommand` to empty and use innerChildren instead.

Reply:
You meant this issue only happened when `spark.sql.runSQLOnFiles`=true? What if `spark.sql.runSQLOnFiles`=false?

Reply:
@maropu This issue affect more aspect, what I point out in the jira is just for the case "table or view not found", actually, this issue may cause many other problems.
The root cause is that the the children of InsertIntoDataSourceDirCommand is empty, thus many analysis rules after InsertIntoDataSourceDirCommand being inserted(In DataSourceAnalysis rule) may not be effective.
I think we can fix it better inside InsertIntoDataSourceDirCommand, but I should first make it clear that why we set it's children to empty, but use innerChildren instead? Is there any PR or issue for that?

Reply:
I think you should investigate it by yourself.  I don't like the current approach too. Appeartly issue is minor but the fix is pretty invasive. I doesn't need to touch SessionStateBuilder side at all.